java-frontend/src/main/java/org/sonar/java/se/SymbolicExecutionVisitor.java

```diff
@@ -64,18 +66,31 @@ public void visitNode(Tree tree) {
   @CheckForNull
   public MethodBehavior execute(MethodTree methodTree) {
     try {
-      MethodBehavior methodBehavior = new MethodBehavior(methodTree.symbol());
-      behaviorCache.add(methodTree.symbol(), methodBehavior);
+      Symbol.MethodSymbol methodSymbol = methodTree.symbol();
       ExplodedGraphWalker walker = egwFactory.createWalker(behaviorCache);
-      methodBehavior = walker.visitMethod(methodTree, methodBehavior);
-      methodBehavior.completed();
-      return methodBehavior;
+      if (methodCanNotBeOverriden(methodSymbol)) {
+        MethodBehavior methodBehavior = new MethodBehavior(methodSymbol);
+        behaviorCache.add(methodSymbol, methodBehavior);
+        methodBehavior = walker.visitMethod(methodTree, methodBehavior);
+        methodBehavior.completed();
+        return methodBehavior;
+      } else {
+        return walker.visitMethod(methodTree);
+      }
     } catch (ExplodedGraphWalker.MaximumStepsReachedException | ExplodedGraphWalker.ExplodedGraphTooBigException | BinaryRelation.TransitiveRelationExceededException exception) {
       LOG.debug("Could not complete symbolic execution: ", exception);
     }
     return null;
   }
 
+  private static boolean methodCanNotBeOverriden(Symbol.MethodSymbol methodSymbol) {
+    if ((((JavaSymbol.MethodJavaSymbol) methodSymbol).flags() & Flags.NATIVE) != 0) {
+      return false;
+    }
+    return !methodSymbol.isAbstract() &&
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)